From 8e8590fe62c5b1c9daceb7588798a96b95df6039 Mon Sep 17 00:00:00 2001
From: MeiChia Chiu <meichia.chiu@mediatek.com>
Date: Fri, 14 Jul 2023 17:19:13 +0800
Subject: [PATCH 30/54] mtk: hostapd: Update parameter_set_count in MU EDCA IE

without this patch, MU EDCA Parameter update count not equal to
WMM Parameter set count.
---
 src/ap/ieee802_11_he.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/src/ap/ieee802_11_he.c
+++ b/src/ap/ieee802_11_he.c
@@ -317,6 +317,9 @@ u8 * hostapd_eid_he_mu_edca_parameter_se
 	edca = (struct ieee80211_he_mu_edca_parameter_set *) pos;
 	os_memcpy(edca, &hapd->iface->conf->he_mu_edca, sizeof(*edca));
 
+	if (hapd->conf->wmm_enabled)
+		edca->he_qos_info = hapd->parameter_set_count % 0xf;
+
 	wpa_hexdump(MSG_DEBUG, "HE: MU EDCA Parameter Set element",
 		    pos, sizeof(*edca));
 
